docs/tutorials/time-series.md

```diff
@@ -0,0 +1,738 @@
+inputs arrive. Such incremental updates include deleting existing
+rows and inserting new rows into the view.  By monitoring these
+updates, the user can observe the most up-to-date results of the
+computation.  However, some applications only need to observe its
+final outputs, i.e., rows that are guaranteed to
+never get deleted or updated.
+
+Consider the following query that computes the sum of all purchases
+for each day.
+
+```sql
+CREATE VIEW daily_total AS
+SELECT
+    TIMESTAMP_TRUNC(ts, DAY) as d,
+    SUM(amount) AS total
+FROM
+    purchase
+GROUP BY
+    TIMESTAMP_TRUNC(ts, DAY);
+```
+
+Suppose that a new `purchase` record arrives every second.  In response,
+Feldera updates the `daily_total` value for the current date,
+deleting the old value and inserting a new one.  The user may observe
+86400 such updates per day.  In some cases this is precisely what the user
+wants--to keep track of the latest value of the aggregate with low latency.
+But if the application only requires the final value of the aggregate at
+the end of the day, handling all the intermediate updates may introduce
+unnecessary overheads. Some applications may not be able to handle
+intermediate updates at all.  Typical reasons for this are:
+
+* **The application is unable to process row deletions**. Some
+    databases and database connectors cannot ingest deletions
+    efficiently or at all.
+
+* **The application cannot handle a large volume of incremental
+    updates.**  As we see in this example, the volume of intermediate
+    updates can be much larger than the final output, potentially
+    overwhelming the application.
+
+* **The application is only allowed to act on the final
+    output**. Consider an application that raises an alarm, sends an
+    email, or performs some other irreversible action based on the
+    output or the view. Such actions should only be taken when the
+    output is final. Handling intermediate outputs can
+    complicate the logic of the application, as the application may
+    not be able to reliably identify and discard intermediate values.
+
+The `emit_final` attribute instructs Feldera to only output final rows
+of a view, as in the following example:
+
+```sql
+CREATE VIEW daily_total
+WITH (
+    'emit_final' = 'd'
+)
+AS
+SELECT
+    TIMESTAMP_TRUNC(ts, DAY) as d,
+    SUM(amount) AS total
+FROM
+    purchase
+GROUP BY
+    TIMESTAMP_TRUNC(ts, DAY);
+```
+
+The `emit_final` annotation causes the view to emit only the rows that
+have a value in the specified column that is before the view's current
+waterline.  Let us insert some records in the `purchase` table and
+observe how this affects the waterlines and the output of the view
+with and without `emit_final` annotations.
+
+| `INSERT INTO purchase`              | `purchase.ts` stream time  |  `purchase.ts` waterline     | `daily_total.d` waterline    | Output without `emit_final`                     | Output with `emit_final`    |
```

Comment:
why don't you make these dates or some other shorter data type so the stuff is easier to read?